File: 01-数据结构/01-数组/c++/LC 移动零.cpp
#include<vector>

using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != 0) continue;
            for (int j = i+1; j < nums.size(); j++) {
                if (nums[j] == 0) continue;
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                break;
            }
        }
    }

    void moveZeroes2(vector<int>& nums) {
        int n = nums.size();
        int left = 0;
        int right = 0;

        while (right < n) {
            if (nums[right] != 0) {
                swap(nums[left], nums[right]);
                left++;
            }
            right++;
        }
    }
};

File: 01-数据结构/01-数组/go/LC 移动零.go
package main

func moveZeroes(nums []int)  {
    for i, v := range nums {
        if v != 0 {
            continue
        }

        for j, v2 := range nums[i:] {
            if v2 == 0 {
                continue
            }
            nums[i], nums[j+i] = nums[j+i], nums[i]
            
            break
        }
    }
}